netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -350,6 +351,14 @@ private void createStream(CreateStreamCommand command, final ChannelPromise prom
     final NettyClientStream stream = command.stream();
     final Http2Headers headers = command.headers();
     stream.id(streamId);
+
+    if (goAwayStatus != null) {
+      // The connection is going away, just terminate the stream now.
+      promise.setFailure(goAwayStatus.asException());
+      stream.transportReportStatus(goAwayStatus, false, new Metadata());
```

Comment:
Yup ... good catch.  Actually, we shouldn't be directly calling transportReportStatus anywhere in this method.  I've tweaked the future handling logic so that it is no longer necessary.